### ISSUE TITLE
scrollable-region-focusable should consider an element's content real content, not its content box

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4535]
The following shouldn't fail the `scrollable-region-focusable` rule:

```html
<style>
  #container {
    width: 300px;
    background: #aff;
    overflow-y: auto;
  }
  p {
    width: 600px;
  }
</style>
<div id="container">
  <p>Contents</p>
</div>
```

The text in this paragraph is fully visible. Axe-core is failing this because its detecting a `p` element who's content box ix much larger than it scrollable parent. Axe shouldn't use the content box here though, it should use the actual size and position of the element's content.

**Comment by padmavemulapati** [comment]
Verified with develop version of axe-core, not seeing `scrollable-regio-focusable` rule failing. 
"The text in this paragraph is fully visible.  axe-core is detecting a `p` element who's content box is much larger than scrollable parent. 

https://github.com/user-attachments/assets/fecaf936-567b-4310-acbf-10f8ee841a6a
